test/build-system-mkosi-wrapper.sh

```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: LGPL-2.1-or-later
+set -eux
+image="$1"
+output="$2"
+shift 2
+mkdir -p "$output"
+mkosi -C .. --image="$image" --format=disk --output-dir="$(readlink -f "$output")" --without-tests -ff "$@" build
```

Comment:
-ff is too extreme here, especially if we're always marking the mkosi target as stale. Let's use `-f` for now. Yes it won't always be correct if new dependencies are added to the packages, but we can deal with that for now until we find a better solution.

Reply:
In general, I would prefer to just inline this script into meson. I don't the `mkdir` is required either as mkosi will create the output dir itself?

Reply:
If we kept the script we _could_ retry if a `-fi` fails, but inlining is good enough for CI.

It's a bit verbose repeating the arguments in 6 different places but I'm hoping we can reduce the number of distinct images needed.

Is a `mkosi clean` enough to fix the intermediate when packages change? After switching over to use journal forwarding we can use `--ephemeral` which will remove the need for a separate output dir.

Reply:
@richardmaw-codethink `mkosi -f clean` will be needed. I'll try and find a solution that allows us to track packages from packaging specs in the cache manifest so this won't be necessary anymore.